elks/arch/i86/drivers/char/config.in

```diff
@@ -11,24 +11,35 @@ mainmenu_option next_comment
 		 8018x		CONFIG_CONSOLE_8018X 	\
 		 Headless       CONFIG_CONSOLE_HEADLESS"	Direct
 	if [ "$CONFIG_CONSOLE_DIRECT" = "y" ]; then
-		bool '  Scancode keyboard driver'	CONFIG_KEYBOARD_SCANCODE y
+		if [ "$CONFIG_ARCH_SWAN" = "y" ]; then
+			choice 'Console font'			\
+				"8x8   CONFIG_CONSOLE_FONT_8X8  \
+				4x8    CONFIG_CONSOLE_FONT_4X8" 4x8
+			bool '  Console bell'    CONFIG_CONSOLE_BELL y
+		else
+			bool '  Scancode keyboard driver'	CONFIG_KEYBOARD_SCANCODE y
+		fi
```

Comment:
I'm OK with this change also, but what we're really doing is adding all sorts of complexity to our mainline configuration, all for a port that's likely not used by many people. This adds long term complexity that's hard to maintain.

I can see that each architecture shouldn't present options that aren't valid for it, but perhaps we should be considering a different solution here: a config.swan.in file specific to the port. In that way, we don't have the mess of if statements all over the place, but just with an include somewhere.

Think elegance: how can we add more ports but maintain and/or build elegeance in ELKS source?

Reply:
Reverted them for now. If a config rework is to be done, it would ideally involve all platforms, not just mine.